packages-exp/auth-exp/src/core/user/reload.test.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+
+import { mockEndpoint } from '../../../test/api/helper';
+import { testUser } from '../../../test/mock_auth';
+import * as fetch from '../../../test/mock_fetch';
+import { Endpoint } from '../../api';
+import {
+  APIUserInfo,
+  ProviderUserInfo
+} from '../../api/account_management/account';
+import { UserInfo } from '../../model/user';
+import { ProviderId } from '../providers';
+import { _reloadWithoutSaving, reload } from './reload';
+
+use(chaiAsPromised);
+use(sinonChai);
+
+const BASIC_USER_INFO: UserInfo = {
+  providerId: ProviderId.FIREBASE,
+  uid: 'uid',
+  email: 'email',
+  displayName: 'displayName',
+  phoneNumber: 'phoneNumber',
+  photoURL: 'photoURL'
+};
+
+const BASIC_PROVIDER_USER_INFO: ProviderUserInfo = {
+  providerId: ProviderId.FIREBASE,
+  rawId: 'uid',
+  email: 'email',
+  displayName: 'displayName',
+  phoneNumber: 'phoneNumber',
+  photoUrl: 'photoURL'
+};
+
+describe('reload()', () => {
+  beforeEach(fetch.setUp);
+  afterEach(fetch.tearDown);
+
+  it('sets all the new properties', async () => {
+    const serverUser: APIUserInfo = {
+      localId: 'localId',
+      displayName: 'displayName',
+      photoUrl: 'photoURL',
+      email: 'email',
+      emailVerified: true,
+      phoneNumber: 'phoneNumber',
+      tenantId: 'tenantId',
+      createdAt: 123,
+      lastLoginAt: 456
+    };
+
+    mockEndpoint(Endpoint.GET_ACCOUNT_INFO, {
+      users: [serverUser]
+    });
+
+    const user = testUser('abc', '', true);
+    await _reloadWithoutSaving(user);
```

Comment:
can we test reload instead of _reloadWithoutSaving? Feels like an anti-pattern to test private helpers instead of the actual interface

Reply:
`_reloadWithoutSaving()` gets used a ton throughout the rest of the SDK (more often than `reload()`). It's "private" in the sense that _other_ packages shouldn't be using it, but we'll be using it a lot